lib/auth/types/openid/OpenId.js

```diff
@@ -121,29 +193,141 @@ export default class OpenId extends AuthType {
+            if (differentAuthHeaderCredentials) {
+                try {
+                    let authResponse = await request.auth.securitySessionStorage.authenticate(differentAuthHeaderCredentials);
+                    return { valid: true, credentials: authResponse.session };
+                } catch (error) {
+                    request.auth.securitySessionStorage.clearStorage();
+                    return { valid: false };
+                }
+            }
+
+            if (session.exp && session.exp <= Math.floor(Date.now() / 1000)) {
+                var refreshedSession = {};
+                if (session.credentials && session.credentials.refresh_token && session.credentials.refresh_token !== null
+                    && session.credentials.refresh_token !== '') {
+                    try {
+                        let body = await this.refreshTokenRequest(session.credentials.refresh_token);
+                        let result = JSON.parse(body);
+                        let id_token = result['id_token'];
+                        let tokenPayload = JSON.parse(Buffer.from(id_token.split('.')[1], 'base64').toString());
+                        refreshedSession.username = tokenPayload['name'];
+                        refreshedSession.credentials = {};
+                        refreshedSession.credentials.authHeaderValue = 'Bearer ' + id_token;
+                        refreshedSession.authType = 'openid';
+                        if (tokenPayload.exp) {
+                            this.sessionKeepAlive = false;
+                            refreshedSession.exp = parseInt(tokenPayload.exp, 10);
+                        }
+                        let {user} = await request.auth.securitySessionStorage.authenticate({
+                            authHeaderValue: 'Bearer ' + id_token,
+                            refresh_token: result['refresh_token']
+                        });
+                        request.state[this.config.get('opendistro_security.cookie.name')] = request._states[this.config.get('opendistro_security.cookie.name')].value ;
+                        return { valid: true, credentials: refreshedSession };
+                    } catch (error) {
+                        this.server.log(['error', 'security'], `Error while refreshing the access token : ${error}`);
+                        request.auth.securitySessionStorage.clearStorage();
+                        return { valid: false };
+                    }
+                } else {
+                        this.server.log(['error', 'security'], `Error while fetching refreshToken : ${error}`);
+                        request.auth.securitySessionStorage.clearStorage();
+                        return { valid: false };
+                    }
+            } else if (!session.exp && this.sessionTTL) {
+                if (!session.expiryTime || session.expiryTime < Date.now()) {
+                    request.auth.securitySessionStorage.clearStorage();
```

Comment:
Why is this removed ? this._cookieValidationError = new SessionExpiredError('Session expired')

Reply:
_cookieValidationError is removed as the code to check for the error is removed. For openId, the check to handle cookieValidation is redirecting the user for a login, but when using with IDP the request is not redirected to IDP as Kibana getting struck as CORS policy comes into effect as it stops any outgoing kibana request to IDP.